OutOfMemoryError when running Model Tests in CI

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

storage

### Describe the bug

Sometimes the model tests fail with an OOM error: 

> 2023-02-25T02:47:00.9801885Z [ERROR] java.lang.OutOfMemoryError: Java heap space
2023-02-25T02:47:01.2779542Z [ERROR] Terminating due to java.lang.OutOfMemoryError: Java heap space

> 2023-02-25T02:47:01.2881118Z [ERROR] Error occurred in starting fork, check output in log
2023-02-25T02:47:01.2881742Z [ERROR] Process Exit Code: 3
2023-02-25T02:47:01.2882072Z [ERROR] Crashed tests:
2023-02-25T02:47:01.2882533Z [ERROR] org.keycloak.testsuite.model.infinispan.CacheExpirationTest
2023-02-25T02:47:01.2883188Z [ERROR]    at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:714)
2023-02-25T02:47:01.2883885Z [ERROR]    at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:311)
2023-02-25T02:47:01.2884579Z [ERROR]    at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:268)


https://github.com/keycloak/keycloak/actions/runs/4267884712/jobs/7429994617

### Version

nightly

### Expected behavior

Tests should complete

### Actual behavior

Tests are flaky

### How to Reproduce?

Seems to randomly fail, hard to reproduce. 

### Anything else?

Increasing the heap size as a first measure.

--- a/testsuite/model/pom.xml
+++ b/testsuite/model/pom.xml
@@ -161,7 +161,7 @@
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
                     <!-- See also https://github.com/wildfly/wildfly-core/blob/7e5624cf92ebe4b64a4793a8c0b2a340c0d6d363/core-feature-pack/common/src/main/resources/content/bin/common.sh#L57-L60 -->
-                    <argLine>@{argLine} -Xmx1024m -XX:+ExitOnOutOfMemoryError -XX:+HeapDumpOnOutOfMemoryError --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED</argLine>
+                    <argLine>@{argLine} -Xmx1536m -XX:+ExitOnOutOfMemoryError -XX:+HeapDumpOnOutOfMemoryError --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED</argLine>
                     <systemPropertyVariables>
                         <!-- keycloak.model.parameters lists parameter classes from 
                              org.keycloak.model.parameters package and determine enabled providers -->

